diff -Nru rpm-4.17.1.1.orig/rpmio/rpmlua.c rpm-4.17.1.1/rpmio/rpmlua.c
--- rpm-4.17.1.1.orig/rpmio/rpmlua.c	Fri Sep  2 12:12:50 2022
+++ rpm-4.17.1.1/rpmio/rpmlua.c	Sat Feb 24 03:12:49 2024
@@ -13,7 +13,6 @@
 
 #include <unistd.h>
 #include <assert.h>
-#include <spawn.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <stdarg.h>
@@ -750,6 +749,30 @@
     exit(luaL_optinteger(L, 1, EXIT_SUCCESS));
 }
 
+static int
+posix_spawnp(pid_t *pid, const char *path,
+	char * const argv[])
+{
+	pid_t p;
+	volatile int error = 0;
+
+	p = vfork();
+	switch (p) {
+	case -1:
+		return (errno);
+	case 0:
+		execvp(path, argv);
+		error = errno;
+		_exit(127);
+	default:
+		if (error != 0)
+			waitpid(p, NULL, WNOHANG);
+		else if (pid != NULL)
+			*pid = p;
+		return (error);
+	}
+}
+
 static int rpm_execute(lua_State *L)
 {
     const char *file = luaL_checkstring(L, 1);
@@ -765,7 +788,7 @@
 	argv[i] = (char *)luaL_checkstring(L, i + 1);
     argv[i] = NULL;
     rpmSetCloseOnExec();
-    status = posix_spawnp(&pid, file, NULL, NULL, argv, environ);
+    status = posix_spawnp(&pid, file, argv);
     free(argv);
     if (status != 0)
 	return pusherror(L, status, "posix_spawnp");
